src/blocks/folder-blocks/infinite-canvas/FilePicker.tsx

```diff
@@ -0,0 +1,43 @@
+import { useState } from "react";
+import { Files, File } from "./index";
+
+export const FilePicker = ({
+  files,
+  onFileSelected,
+}: {
+  files: Files;
+  onFileSelected: (file: File) => void;
+}) => {
+  // there must be a better way to close the dropdown!
+  const [iteration, setIteration] = useState(0);
```

Comment:
yeah, really not ideal, but this was the best way I found to close the open picker (because it's re-rendered) when an item is selected:
https://github.com/githubnext/blocks-examples/blob/15aa99611048457dd8b5baf49f6e8fa707b6b1ac/src/blocks/folder-blocks/infinite-canvas/FilePicker.tsx#L31

`detail` elements are, by default, uncontrolled components, because the browser handles opening/closing. An alternative would be to completely replace the open/closing logic by setting the `open` attribute and handling onClick events, but I think re-rendering is the least evil, since it's a pretty lightweight element. I could definitely be wrong there, though!

Reply:
> `detail` elements are, by default, uncontrolled components, because the browser handles opening/closing.

ah ok this is the part I was missing. thank you!